sopel/modules/url.py

```diff
@@ -303,14 +303,20 @@ def title_auto(bot, trigger):
     if re.match(bot.config.core.prefix + r'\S+', trigger):
         return
 
-    # Avoid fetching known malicious links
-    if 'safety_cache' in bot.memory and trigger in bot.memory['safety_cache']:
-        if bot.memory['safety_cache'][trigger]['positives'] > 1:
-            return
-
-    urls = web.search_urls(
+    unchecked_urls = web.search_urls(
         trigger, exclusion_char=bot.config.url.exclusion_char, clean=True)
 
+    urls = []
+    for url in unchecked_urls:
+        # Avoid fetching known malicious links
+        if "safety_cache" in bot.memory and url in bot.memory["safety_cache"]:
+            if bot.memory["safety_cache"][url]["positives"] > 0:
+                continue
+        netloc = urlparse(url).netloc.lower()
+        if "safety_cache_local" in bot.memory and netloc in bot.memory["safety_cache_local"]:
```

Comment:
Same simplification here:

```suggestion
        if netloc in bot.memory.get("safety_cache_local", {}):
```